modules/tracker/controllers/ScalarController.php

```diff
@@ -50,18 +51,24 @@ public function detailsAction()
         /** @var Tracker_ScalarDao $scalarDao */
         $scalarDao = $this->Tracker_Scalar->load($scalarId);
 
+        /** @var Tracker_SubmissionDao $submissionDao */
+        $submissionDao = $this->Tracker_Submission->load($scalarDao->getSubmissionId());
+
         if ($this->Tracker_Scalar->policyCheck($scalarDao, $this->userSession->Dao, MIDAS_POLICY_READ) === false
         ) {
             throw new Zend_Exception('The scalar does not exist or you do not have the necessary permission', 403);
         }
 
+        // TODO(cpatrick): This may be a performance issue.
         $this->view->isAdmin = $this->Tracker_Scalar->policyCheck($scalarDao, $this->userSession->Dao, MIDAS_POLICY_ADMIN);
+
         $this->view->scalar = $scalarDao;
-        $this->view->extraParams = $scalarDao->getParams();
-        $this->view->extraUrls = json_decode($scalarDao->getExtraUrls(), true);
+        $this->view->submission = $submissionDao;
+        $this->view->extraParams = $submissionDao->getParams();
+        $this->view->extraUrls = json_decode($submissionDao->getExtraUrls(), true);
 
-        $revisionUrl = $scalarDao->getTrend()->getProducer()->getRevisionUrl();
-        $producerRevision = $scalarDao->getProducerRevision();
+        $revisionUrl = $submissionDao->getProducer()->getRevisionUrl();
+        $producerRevision = $submissionDao->getProducerRevision();
```

Comment:
The revisionUrl has a placeholder to insert the actual revision string. It's attached to the producer itself. This makes sense, I think.